### PR TITLE
Implement fstat.

### PR DESCRIPTION
**Pull request by zvin**
This is required for lkl.fs.readFile

Change-Type: patch

### CODE REVIEW

**Comment by zvin** [comment]
I am starting to use node-lkl in a branch of resin-image-maker.
I need to use lkl.fs.readFile and it requires implementing fstat.

File: lib/binding.js [hunk 1]
```diff
@@ -59,8 +67,15 @@ exports.fdatasync = function() {
 	throw new Error('Unimplemented');
 }
 
-exports.fstat = function() {
-	throw new Error('Unimplemented');
+exports.fstat = function(fd, req) {
+	const buf = new Buffer(STRUCT_LKL_STAT_SIZE);
```
Comment (petrosagg) [inline lib/binding.js]: `new Buffer` as a way to create a buffer is deprecated. You can use either `Buffer.alloc()` or `Buffer.allocUnsafe()`

File: src/node_lkl.cc [hunk 2]
```diff
@@ -130,3 +130,58 @@ NAN_METHOD(parseDirent64) {
 	}
 	info.GetReturnValue().Set(result);
 }
+
+NAN_METHOD(sizeOfStructLklStat) {
+	auto size = Nan::New<v8::Uint32>(
+		static_cast<uint32_t>(sizeof(struct lkl_stat))
```
Comment (petrosagg) [inline src/node_lkl.cc]: Out of curiosity, what type does `sizeof` return?

File: src/node_lkl.cc [hunk 3]
```diff
@@ -130,3 +130,58 @@ NAN_METHOD(parseDirent64) {
...
+}
+
+
+NAN_METHOD(parseLklStat) {
+	// Args:
+	// * Buffer buffer
+	// * function Stats
+	// Parses buffer (which should be a struct lkl_stat*) and returns a Stats
+	// object.
+	auto *s = (struct lkl_stat*) node::Buffer::Data(info[0]);
```
Comment (petrosagg) [inline src/node_lkl.cc]: why not a `reinterpret_cast<struct lkl_stat*>(node::Buffer::Data(info[0]))`? Or you can even omit the struct and do `reinterpret_cast<lkl_stat*>(node::Buffer::Data(info[0]))`

File: src/node_lkl.cc (hunk 2 above)
Comment (zvin) [inline src/node_lkl.cc]: A size_t (long unsigned int).
This is what the compiler says if I don't cast
```
../node_modules/nan/nan_new.h:208:30: error: call of overloaded ‘New(long unsigned int&)’ is ambiguous
   return imp::Factory<T>::New(arg0);
                              ^
In file included from ../node_modules/nan/nan_new.h:189:0,
                 from ../node_modules/nan/nan.h:196,
                 from ../src/node_lkl.h:1,
                 from ../src/node_lkl.cc:6:
../node_modules/nan/nan_implementation_12_inl.h:169:1: note: candidate: static Nan::imp::FactoryBase<v8::Uint32>::return_t Nan::imp::Factory<v8::Uint32>::New(int32_t)
 Factory<v8::Uint32>::New(int32_t value) {
 ^
../node_modules/nan/nan_implementation_12_inl.h:175:1: note: candidate: static Nan::imp::FactoryBase<v8::Uint32>::return_t Nan::imp::Factory<v8::Uint32>::New(uint32_t)
 Factory<v8::Uint32>::New(uint32_t value) {
```

File: src/node_lkl.cc (hunk 3 above)
Comment (zvin) [inline src/node_lkl.cc]: Good idea, fixed.

File: lib/binding.js (hunk 1 above)
Comment (zvin) [inline lib/binding.js]: fixed

**Pull request merged**